src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -54,7 +57,7 @@ protected override void Initialize()
             var method = compilerFailFast.GetMethod(nameof(FailFast.OnFatalException), BindingFlags.Static | BindingFlags.NonPublic);
             property.SetValue(null, Delegate.CreateDelegate(property.PropertyType, method));
 
-            var componentModel = (IComponentModel)this.GetService(typeof(SComponentModel));
+            var componentModel = (IComponentModel)await GetServiceAsync(typeof(SComponentModel)).ConfigureAwait(true);
             _workspace = componentModel.GetService<VisualStudioWorkspace>();
```

Comment:
If `VisualStudioWorkspace` is as heavy as it sounds, maybe you should switch to a background thread before querying MEF for it. Otherwise you could end up loading a bunch of assemblies on the UI thread, which wouldn't be good.

Reply:
Many of these types are UI affinitized, so this is something we may visit in the future but for now I preserved the original behavior.

Reply:
The MEF fetch itself *should* be safe; the UI thread stuff was all moved to VisualStudioWorkspaceImpl.GetProjectTrackerAndInitializeIfNecessary(). If that *isn't* safe, well, then somebody broke it since we did that refactoring.